### PR TITLE
Feature/end lession option

### PR DESCRIPTION
**Pull request by arrra**
User can end lession

### CODE REVIEW

File: client/src/assets/partials/lesson.html [hunk 1]
```diff
@@ -66,6 +68,10 @@ <h3>Level {{selectedLevel}}</h3>
 
 <div class="divider"></div>
 
+<div class="lesson-end" ng-show="lesson.state === 'Running'">
```
Comment (mrap) [inline client/src/assets/partials/lesson.html]: Don't compare vs strings when we have enum values - strings are very brittle.
Better yet, `lesson` can provide a helper method like `lesson.isRunning()`

File: client/src/assets/stylesheets/partials/_lesson.scss [hunk 2]
```diff
@@ -16,6 +16,15 @@ $letterSize: 8em;
   background: $progress-color;
 }
 
+.lesson-end {
+  text-align: center;
+  margin-top: 1rem;
+}
+
+.lesson-end .btn {
+  background-color: blue;
```
Comment (mrap) [inline client/src/assets/stylesheets/partials/_lesson.scss]: This is a low-priority and infrequent user action, we should be more subtle about it to keep the user's focus on the task at hand: the lesson itself.

**Review by mrap — CHANGES_REQUESTED** [review]

File: client/src/assets/partials/lesson.html (hunk 1 above)
Comment (arrra) [reply client/src/assets/partials/lesson.html]: Helper method makes sense.

File: client/src/assets/stylesheets/partials/_lesson.scss (hunk 2 above)
Comment (arrra) [reply client/src/assets/stylesheets/partials/_lesson.scss]: Got it!. I just realized we don't use this code as well.

File: client/src/assets/js/app/typing-lesson.js [hunk 3]
```diff
@@ -103,7 +103,16 @@
...
-      }
+      },
+      endLesson: function() {
+        this.markCompleted();
+      },
+      accuracyLabel: function() {
+        return (this.accuracy) ? `${this.accuracy}%` : '-';
+      },
+      isRunning: function() {
+        return this.state === LevelStates.Running;
```
Comment (mrap) [inline client/src/assets/js/app/typing-lesson.js]: For clarity, we should wrap conditionals in `()` if we use the result as a value:

`return (this.state === LevelStates.Running);`

File: client/src/assets/partials/lesson.html [hunk 4]
```diff
@@ -47,6 +47,8 @@ <h3>Level {{selectedLevel}} not yet mastered</h3>
         {{lesson.combosLeft}} Combos Left
       </div>
     </div>
+
```
Comment (mrap) [inline client/src/assets/partials/lesson.html:50]: Unnecessary addition.

**Review by mrap — CHANGES_REQUESTED** [review]

**Review by mrap — APPROVED** [review]

**Pull request merged**